dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/UsersServiceImpl.java

```diff
@@ -203,23 +198,23 @@ public User createUser(String userName,
                            String phone,
                            String queue,
                            int state) {
-        User user = new User();
         Date now = new Date();
+        checkUserParams(userName, userPassword, email, phone);
+
+        User tempUser = userMapper.queryByUserNameAccurately(userName);
+        if (tempUser != null) {
+            throw new ServiceException(Status.NAME_EXIST, userName);
+        }
+
+        User user = new User(userName, userPassword, email, phone, tenantId, state);
 
-        user.setUserName(userName);
-        user.setUserPassword(EncryptionUtils.getMd5(userPassword));
-        user.setEmail(email);
-        user.setTenantId(tenantId);
-        user.setPhone(phone);
-        user.setState(state);
         // create general users, administrator users are currently built-in
         user.setUserType(UserType.GENERAL_USER);
-        user.setCreateTime(now);
-        user.setUpdateTime(now);
         if (StringUtils.isEmpty(queue)) {
             queue = "";
         }
         user.setQueue(queue);
+        user.setCreateTime(now);
```

Comment:
can we add `createtime` to the class `User` constructor? we can add multiple constructor to different behavior

```java
    // for new user 
    public User(String userName, String userPassword, String email, String phone, Integer tenantId, Integer state) {
         Date now = new Date();
         this.userName = userName;
         this.userPassword = EncryptionUtils.getMd5(userPassword);
         this.email = email;
         this.phone = phone;
         this.tenantId = tenantId;
         this.state = state;
         this.updateTime = now;
         this.createTime = now;
     }

    // for update user, we keep the create 
    public User(int userId, String userName, String userPassword, String email, String phone, Integer tenantId, Integer state, Date createTime) {
        this.User(userName, userPassword, email, phone, tenantId, state)         
        this.id = userId; 
        this.createTime = createTime;
     }
```